### PR TITLE
Changes for new Collection creation on transformation

### PR DESCRIPTION
**Pull request by phillip-kruger**
Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

Changes:
1) Only create new collection if necessary (there is transformation on the objects)
2) Make sure that if we have to create a new collection, that we use the type from the method signature and not the object's type.
3) Centralized classloading.

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java [hunk 1]
```diff
@@ -38,59 +45,69 @@
...
+                                // Let's try this class classloader.
+                            }
+                        }
+                        Class<?> c = Class.forName(className, false, Classes.class.getClassLoader());
+                        loadedClasses.put(className, c);
+                        return c;
+                    });
+                } catch (PrivilegedActionException pae) {
+                    throw new ClassloaderException("Can not load class [" + className + "]", pae);
+                }
```
Comment (t1) [inline implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java:79]: maybe this block could be extracted into a method, while the `loadedClasses.put(className, c);` happens here.

File: implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java [hunk 2]
```diff
@@ -38,59 +45,69 @@
...
  */
 public class Classes {
+    private static final Map<String, Class> loadedClasses = new HashMap<>();
 
     private Classes() {
     }
 
+    public static Class<?> loadClass(String className) {
+
+        if (isPrimative(className)) {
```
Comment (t1) [inline implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java:55]: I think this is a type: should be 'primitive'

Comment (phillip-kruger) [reply implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java:55]: Thanks :) Fixing

File: implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java (hunk 1 above)
Comment (phillip-kruger) [reply implementation/src/main/java/io/smallrye/graphql/bootstrap/Classes.java:79]: Yes cool. loadedClasses.put can actually also happen in the method

**Pull request merged**